File: lessons/day_06/netcore/src/GraphQLSample/Models/Input/InputPersonCreate.cs
using GraphQLNetCore.Models.Abstractions;
using GraphQLNetCore.Models.Enums;
using System;
using System.Collections.Generic;

namespace GraphQLNetCore.Models.Input
{
   public class InputPersonCreate : IInputPersonCreate<Contact>
   {
      public string Name { get; set; }
      public string Surname { get; set; }
      public string Email { get; set; }
      public int? Age { get; set; }
      public EyeColor? EyeColor { get; set; }
      public List<int> Friends { get; set; }
      public List<int> Skills { get; set; }
      public int? FavSkill { get; set; }

      public Contact ToPerson()
      {
         return new Contact() {
            Age = Age ?? 0,
            Email = Email ?? string.Empty,
            EyeColor = EyeColor,
            FavSkillId = FavSkill,
            Name = Name,
            Surname = Surname ?? string.Empty
         };
      }
   }
}

File: lessons/day_05/java/src/main/java/com/example/DemoGraphQL/input/InputPersonCreate.java
package com.example.DemoGraphQL.input;

import com.example.DemoGraphQL.model.EyeColor;

import java.util.List;

public record InputPersonCreate (
    String name,
    String surname,
    String email,
    Integer age,
    EyeColor eyeColor,
    Long favSkill,
    List<Long> friends,
    List<Long> skills
){}
